src/main/java/backend/SearchAlg.java

```diff
@@ -0,0 +1,93 @@
+package backend;
+
+import utils.Logging.MiniMLLogger;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+/**
+ * Explore the parameter space of an algorithm.
+ */
+public interface SearchAlg {
+    ArrayList<WrappedParamFinal> getNextParamSet(ParameterIFace parameterSet);
+}
+
+/**
+ * Class implements truly random search of parameter space.
+ */
+class randomSearch implements SearchAlg {
+    Dispatcher dsp;
+    Random random = new Random();
+
+    /**
+     * Build our search algorithm and make it know who holds it.
+     * This allows it to access other members of the dispatcher (like parameterIFace instances)
+     *
+     * @param owner
+     */
+    public randomSearch(Dispatcher owner){
+        dsp = owner;
+    }
+
+    /**
+     * Generate a random set of parameters for a model.
+     *
+     * Looks over all possible parameters in the ParameterIFace and assign values in their appropriate ranges.
+     * Also handles flags that do not accept values (we call them 'empty' flags)
+     *
+     * @param parameterSet
+     * @return an ArrayList of WrappedParamFinal to be unpacked and sent to a model.
+     */
+    public ArrayList<WrappedParamFinal> getNextParamSet(ParameterIFace parameterSet){
+        ArrayList<WrappedParamFinal> outParams = new ArrayList<WrappedParamFinal>();
+        ArrayList<WrappedParam> inParams = parameterSet.getParameters();
+        for(WrappedParam p : inParams){
+            switch(p.getType()) {
+                case "int": {   //generate a random ranged int for this parameter
```

Comment:
I don't think an enumeration is going to make this any cleaner; the problem is that some of our parameters are bounded (e.g. 1<= n <= 10), some are binary (true/false, 0/1, etc), some have no value (-C in linear regression), etc.

All we can do is push the hurt around.